src/dropdown/Dropdown.js

```diff
@@ -115,10 +133,85 @@ export default class Dropdown extends React.Component {
 
   focus = () => { this.select.focus(); }
 
+  optionGroupRenderer = ( title ) => {
+    if (title === 'undefined') {
+      return null;
+    }
+    else {
+      return (
+        <FlexView className='Select-option-group'>
+          {this.props.optionGroupRenderer(title)}
+        </FlexView>
+      );
+    }
+  }
+
+  _menuRenderer = ({
+    focusedOption,
+    instancePrefix,
+    onFocus,
+    onSelect,
+    optionClassName,
+    optionComponent,
+    optionRenderer,
+    options,
+    valueArray,
+    valueKey
+  }) => {
+
+    const Option = optionComponent;
+    const propertyName = this.props.groupByKey;
+    const sortedOptions = sortBy(options, (option) => {
```

Comment:
Pay attention here: you're changing the order in which options are rendered, but you're passing them to the underlying `react-select` keeping the original sorting. When you move between options using arrow keys, options are focused in the order you passed them to the `react-select` and not in the order they're rendered:

![](https://i.gyazo.com/e3b78114a26c6bd07a314f4b39af212d.gif)

I suggest to sort them  in `getLocals` and not here.